presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderJoins.java

```diff
@@ -117,14 +117,14 @@ public Result apply(JoinNode joinNode, Captures captures, Context context)
     {
         // We check that join distribution type is absent because we only want to do this transformation once (reordered joins will have distribution type already set).
         MultiJoinNode multiJoinNode = toMultiJoinNode(joinNode, context.getLookup(), JOIN_LIMIT);
-
         JoinEnumerator joinEnumerator = new JoinEnumerator(
                 context.getSession(),
                 context.getCostProvider(),
                 costComparator,
                 context.getIdAllocator(),
                 multiJoinNode.getFilter(),
-                context.getLookup());
+                context.getLookup(),
+                context);
```

Comment:
It seems redundant to pass the whole `context` object along with a bunch of objects which are accessible from it.